### ISSUE TITLE
Token field empty

### ISSUE THREAD

**Issue by malago86** [extiverse/bazaar#135]
### Steps to reproduce

1.  Just installed with composer and activated the extension
2.  The token field is empty
(3.  Also, it's not working with translations, I have a different default language and all the texts are shown as variable names.)

### Expected behaviour

Token field filled with the corresponding token, I tried to reinstall the extension but nothing.

### Actual behaviour

Token field is empty, can't install new extensions.

### Configuration

**Operating system:** Ubuntu 18.04

**Version of Bazaar:** 0.4.0
**Version of Flarum:** beta.9
**Other extensions:** 
+------------------------------+---------------+--------+
| Flarum Extensions            |               |        |
+------------------------------+---------------+--------+
| ID                           | Version       | Commit |
+------------------------------+---------------+--------+
| flarum-approval              | v0.1.0-beta.8 |        |
| flarum-bbcode                | v0.1.0-beta.8 |        |
| flarum-emoji                 | v0.1.0-beta.8 |        |
| flarum-lang-english          | v0.1.0-beta.9 |        |
| flarum-flags                 | v0.1.0-beta.9 |        |
| flarum-likes                 | v0.1.0-beta.9 |        |
| flarum-lock                  | v0.1.0-beta.9 |        |
| flarum-markdown              | v0.1.0-beta.9 |        |
| flarum-mentions              | v0.1.0-beta.9 |        |
| flarum-statistics            | v0.1.0-beta.9 |        |
| flarum-sticky                | v0.1.0-beta.9 |        |
| flarum-subscriptions         | v0.1.0-beta.9 |        |
| flarum-suspend               | v0.1.0-beta.9 |        |
| flarum-tags                  | v0.1.0-beta.9 |        |
| darkfoxdeveloper-spanish     | v1.0.4        |        |
| v17development-seo           | 1.0.1         |        |
| fof-formatting               | 0.1.2         |        |
| fof-pages                    | 0.2.6         |        |
| fof-bbcode-details           | 0.1.0         |        |
| clarkwinkelmann-emojionearea | 0.2.0         |        |
| extiverse-bazaar             | 0.4.0         |        |
| flagrow-analytics            | 0.8.0         |        |
+------------------------------+---------------+--------+

</details>

**Comment by luceos** [comment]
Sorry totally missed this, has this been resolved already?

**Comment by malago86** [comment]
No, still empty!

Actually, I just made a fresh installation of Flarum from scratch in a different directory and database and only installed Bazaar. The token field is empty and I can't connect to the flagrow account (it only opens an about:blank page).

Let me know what information you need, I really want to use Bazaar.

**Comment by xprilion** [comment]
Same issue, token field empty!

**Comment by malago86** [comment]
Upgraded to beta 10 and still no token, I tried to reactivate the extension too, no luck :(

**Comment by malago86** [comment]
> Sorry totally missed this, has this been resolved already?

Upgraded to beta 10 and still no token, I tried to reactivate the extension too, no luck :(

**Comment by luceos** [comment]
Looks like a bug with beta 10 then, I will take a look. Thanks.

**Comment by luceos** [comment]
This is odd, I just tested this locally and it works fine.. Do any of you see anything in the log file? Did you try flushing cache `php flarum cache:clear`?

**Comment by kevinpetit** [comment]
I have the same issue, with the same errors.

**Comment by luceos** [comment]
Me too doesn't help. Post your flarum info output and server information please.

**Comment by kevinpetit** [comment]
spellborn_forum@alesia:~/public_html_flarum$ ./flarum info
Flarum core 0.1.0-beta.10
PHP version: 7.3.9-1+ubuntu18.04.1+deb.sury.org+1
Loaded extensions: Core, date, libxml, openssl, pcre, zlib, filter, hash, pcntl, Reflection, SPL, sodium, session, standard, mysqlnd, PDO, xml, apcu, bcmath, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, gmp, iconv, igbinary, intl, json, ldap, exif, msgpack, mysqli, odbc, pdo_mysql, PDO_ODBC, apc, posix, readline, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Phar, memcached, Zend OPcache
+----------------------+----------------+--------+
| Flarum Extensions    |                |        |
+----------------------+----------------+--------+
| ID                   | Version        | Commit |
+----------------------+----------------+--------+
| flarum-approval      | v0.1.0-beta.8  |        |
| flarum-bbcode        | v0.1.0-beta.8  |        |
| flarum-emoji         | v0.1.0-beta.10 |        |
| flarum-lang-english  | v0.1.0-beta.10 |        |
| flarum-flags         | v0.1.0-beta.9  |        |
| flarum-likes         | v0.1.0-beta.9  |        |
| flarum-lock          | v0.1.0-beta.9  |        |
| flarum-markdown      | v0.1.0-beta.10 |        |
| flarum-mentions      | v0.1.0-beta.10 |        |
| flarum-statistics    | v0.1.0-beta.9  |        |
| flarum-sticky        | v0.1.0-beta.9  |        |
| flarum-subscriptions | v0.1.0-beta.9  |        |
| flarum-suspend       | v0.1.0-beta.9  |        |
| flarum-tags          | v0.1.0-beta.10 |        |
| extiverse-bazaar     | 0.4.0          |        |
+----------------------+----------------+--------+
Base URL: https://forum.spellborn.org
Installation path: /srv/www/forum.spellborn.org/public_html_flarum
Debug mode: off

Runs on Ubuntu 18.04, Nginx 1.14.0. PHP runs in it's own PHP-FPM pool, every file is owned by the user of the site.

**Comment by malago86** [comment]
Still waiting for an answer, the problem seems to be happening to other people too: no token, empty page when trying to connect, my IP is static, DNS are working, bazaar page does show extensions even with an empty token.

**Comment by luceos** [comment]
Can you also post php and database version. Also check whether the settings table has a value for the key named flagrow.bazaar.api_token.

**Comment by malago86** [comment]
PHP: 5.6.37-1+ubuntu18.04.1+deb.sury.org+1
MYSQL: 5.7.27-0ubuntu0.18.04.1

The settings table only has one key with bazaar in its name: 
`flagrow.bazaar.connected | 0`

I hope that helps!

**Comment by luceos** [comment]
The exception shown isn't very explanatory, I think in order to solve this for you and many others it might be wise to give better feedback. I'll whip up a solution coming week.

**Comment by clarkwinkelmann** [comment]
It looks like the extension checks if `flagrow-bazaar` is enabled, even in the new `extiverse-bazaar`. Looks like it could be a reason for this.

https://github.com/extiverse/bazaar/blob/71c9fe35ccfdc77e1c740945d63a33799c3ac9b1/src/Listeners/BazaarEnabled.php#L65

I don't even know why there is a check for extension enabled ? That code should only run if the extension is enabled anyway.

**Comment by luceos** [comment]
@clarkwinkelmann oh 🤦‍♂ great find!

**Comment by luceos** [comment]
@malago86 @xprilion @kevinpetit please upgrade to 0.4.1, let me know when this issue can be closed.

**Comment by malago86** [comment]
Hi @luceos, the token is now showing up!

However, I am still having problems installing extensions from Bazaar, it says that there was an error and I can't see any update in the flarum logs. I thought it was related to the absence of token but it looks like there is something else that is avoiding extensions to be installed.

Also, if I click on connect to flagrow it gives me a blank page too.

I tried reenabling the extension but I get the same results.

TLDR: Token is showing up but Bazaar is still unusable.

**Comment by luceos** [comment]
@malago86 can you check History in Bazaar. In addition the processes might time out, you could set up cron processing, instructions for that are in the settings of Bazaar.
